src/gsi/read_radar.f90

```diff
@@ -907,6 +907,7 @@ subroutine read_radar(nread,ndata,nodata,infile,lunout,obstype,twind,sis,hgtl_fu
      end do superobs
 
      close(lnbufr)     ! A simple unformatted fortran file should not be mixed with a bufr I/O
+     nread=nsuper2_kept
```

Comment:
since there is a QC for level2 rw with VAD wind, nsuper2_in is not the actual number in setuprw. nread should be the sum of nsuper2_kept, nsuper2_5_kept and nsuper3_kept. However, since there is no level2.5 and level2 data, nread is set to nsuper_level2_kept.

Reply:
> since there is a QC for level2 rw with VAD wind, nsuper2_in is not the actual number in setuprw. nread should be the sum of nsuper2_kept, nsuper2_5_kept and nsuper3_kept. However, since there is no level2.5 and level2 data, nread is set to nsuper_level2_kept.

According to the convention from tldpbufr and tldplrso, nread should be the total # after read in. Not the used data after QC. (e.g. L2252-2264)

Reply:
Agree. If-condition is removed. Also in TDR block, "else if (trim(infile) == 'tldplrbufr' ) " is added to cover the two types of TDR files.